### PR TITLE
handle service feeds by bot id 

### PR DESCRIPTION
**Pull request by GuillaumeDSM**

### CODE REVIEW

File: octobot_services/api/service_feeds.py [hunk 1]
```diff
@@ -16,37 +16,27 @@
...
-def get_service_feed(service_feed_class) -> AbstractServiceFeed:
+def get_service_feed(service_feed_class, bot_id) -> AbstractServiceFeed:
     try:
-        return service_feed_class.instance()
+        return ServiceFeeds.instance().get_service_feed(bot_id, service_feed_class.get_name())
     except TypeError:
         raise RuntimeError(f"can't get {service_feed_class} instance: service feed has not been properly created yet")
 
 
-def get_backtesting_service_feed(service_feed_class) -> AbstractServiceFeed:
```
Comment (Herklos) [inline octobot_services/api/service_feeds.py:32]: :+1:

File: octobot_services/api/service_feeds.py [hunk 2]
```diff
@@ -16,37 +16,27 @@
...
-
 async def start_service_feed(service_feed: AbstractServiceFeed, backtesting_enabled: bool) -> bool:
     return await ServiceFeedManager.start_service_feed(service_feed, backtesting_enabled)
 
 
 async def stop_service_feed(service_feed: AbstractServiceFeed) -> None:
     await ServiceFeedManager.stop_service_feed(service_feed)
+
+
+async def clear_bot_id_feeds(bot_id: str) -> None:
```
Comment (Herklos) [inline octobot_services/api/service_feeds.py:41]: :+1:

File: octobot_services/service_feeds/service_feeds.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+from octobot_commons.singleton.singleton_class import Singleton
+
+
+class ServiceFeeds(Singleton):
```
Comment (Herklos) [inline octobot_services/service_feeds/service_feeds.py:19]: Can you add some typing here please ? :wink:

File: octobot_services/service_feeds/service_feed_factory.py [hunk 4]
```diff
@@ -18,17 +18,26 @@
...
         self.config = config
         self.main_async_loop = main_async_loop
+        self.bot_id = bot_id
 
     @staticmethod
-    def get_available_service_feeds() -> list:
-        return get_all_classes_from_parent(AbstractServiceFeed)
+    def get_available_service_feeds(in_backtesting: bool) -> list:
+        feeds = get_all_classes_from_parent(AbstractServiceFeed)
+        if in_backtesting:
```
Comment (Herklos) [inline octobot_services/service_feeds/service_feed_factory.py:34]: What will be the use of `in_backtesting` param ?

Comment (GuillaumeDSM) [reply octobot_services/service_feeds/service_feed_factory.py:34]: To know if the service feeds to be created should be the backtesting versions of the actual service feeds or the regular version

Comment (Herklos) [reply octobot_services/service_feeds/service_feed_factory.py:34]: Oh yes you're  right

**Pull request merged**